wordpress-mobile/WordPress-Android
# Show author filter on pages view

Addressed part of #10726 

This PR is number two in a series that will introduce author filtering on the pages view. I am requesting that this PR be merged into the feature branch. The feature branch was updated with the latest develop branch before starting this subtask.

This PR implements the UI needed to show the author filter selector on the Pages view

**Implementation Details**

- Created `PagesAuthorFilterUIState` to help manage the UI state for the author filter selector. 
- Added MutableLiveData properties for
    - authorSelectionUpdated
    - authorUiState
- Added `updateAuthorFilterSelection` to handle the itemClick on the author filter. It updates the filter with the new selection, updates appPrefs, and updates the viewState pager change, so the PageList knows that it needs to update the shown list to reflect the new filter.
- Added `isFilteringByAuthorSupported` which checks if the filtering by author is supported for this user/site. This follows the same rules as posts and calypso as of 11/2019.
- Added `updateViewStateTriggerPagerChange`  to handle updating the _authorUiState.value
- Updated PagesViewModelTests and added a couple more

Closed | Open 
--------|-------
<kbd><img src="https://user-images.githubusercontent.com/506707/84171956-65793c00-aa49-11ea-84b6-03d3778bbecb.png" width=320></kbd>|<kbd><img src="https://user-images.githubusercontent.com/506707/84171961-6a3df000-aa49-11ea-917f-75b023554972.png" width=320></kbd>


**Notes**

- The data filtering will be implemented in the next PR.
- The Posts feature also shows an author filter and saves state in App prefs. I implemented Pages to use the same appPrefs property, so the filter selection is maintained across pages to posts. **Would love a second opinion on this behavior**.
- There is a “todo” placeholder in `PagesViewModel` . Setting up tracking will occur in a future PR.


To test:

    (A) 
    - Launch WPAndroid
    - Open a wp.com site that has multiple page authors and you are one of them
    - Navigate to Pages
    - Tap on author selector, toggle through “everyone” and “me” 
    - Ensure avatars show as expected
    - Ensure “everyone” and “me” are spelled correctly
    (B)
    - Launch WPAndroid
    - Open a wp.com site that has a single page author (you)
    - Navigate to Pages
    - The author filter should not be showing
    (C)
    - Launch WPAndroid
    - Open a non wp.com
    - Navigate to Pages
    - The author filter should not be showing

PR submission checklist:
